File: src/main/java/io/pivotal/kotlindemo/java/clients/ClientRepository.java
package io.pivotal.kotlindemo.java.clients;

import java.util.List;
import java.util.Optional;

public interface ClientRepository {
    Client create(CreateClientRequest createClientRequest);
    List<Client> findAll();
    Client findById(Long id);
    List<Client> findByLastName(String lastName);
    Optional<Client> findByEmail(String email);
}

File: src/main/kotlin/io/pivotal/kotlindemo/kotlin/clients/ClientRepository.kt
package io.pivotal.kotlindemo.kotlin.clients

interface ClientRepository {
    fun create(createClientRequest: CreateClientRequest): Client
    fun findAll(): List<Client>
    fun findById(id: Long): Client
    fun findByLastName(lastName: String): List<Client>
    fun findByEmail(email: String): Client?
}
